norm_supervisor/norms/abstract.py

```diff
@@ -0,0 +1,53 @@
+from abc import ABC, abstractmethod
+from highway_env.envs.common.action import Action
+from highway_env.vehicle.kinematics import Vehicle
+
+class AbstractNorm(ABC):
+    """Abstract base class for norms.
+
+    Note that these norms may rely on the underlying vehicle model to evaluate violations.
+    """
+    def __init__(self, weight: int, violating_actions: list[Action]) -> None:
+        """Initialize the norm with a weight and a list of potentially violating actions.
+
+        :param weight: the norm weight, used for prioritization.
+        :param violating_actions: list of potentially norm-violating actions.
+        """
+        self.weight            = weight
+        self.violating_actions = violating_actions
+
+    @abstractmethod
+    def evaluate_criterion(self, vehicle: Vehicle, *args, **kwargs) -> float:
```

Comment:
```suggestion
    def evaluate_criteria(self, vehicle: Vehicle, *args, **kwargs) -> float:
```

norms.py uses "criteria" not criterion

Reply:
Good catch